### ISSUE TITLE
Enhance coala-html and integrate into coala core

### ISSUE THREAD

**Issue by li-boxuan** [coala/projects#695]
https://github.com/coala/coala-html is a nice project, but it is an angular application which limits its user scenario and popularity. Users have to install and run an angular application to see the results. What we could possibly do is to make use of coala-html, enhance it and integrate it with coala core. Maybe as a coala CLI option, e.g. '--generate_report'. When using that option, static web pages will be generated.

https://github.com/pyta-uoft/pyta is a project initiated by the University of Toronto. It looks like a minified version of coala, but is designed to be beginner-friendly. After running the tool, it does not generate reports via CLI, but via static web pages. The attached zip file is a sample report generated by pyta.

[PyTA Error Report.html.zip](https://github.com/coala/projects/files/2648048/PyTA.Error.Report.html.zip)

**Comment by li-boxuan** [comment]
@KVGarg Maybe you can write some of your thoughts here.

**Comment by li-boxuan** [comment]
I guess `integrate into coala core` is not necessary once https://github.com/coala/cEPs/pull/140 is done. But that sounds like a huge project, so I have no idea if this project is viable.

**Comment by KVGarg** [comment]
> I guess `integrate into coala core` is not necessary once [coala/cEPs#140](https://github.com/coala/cEPs/pull/140) is done. But that sounds like a huge project, so I have no idea if this project is viable.

Ok and I will soon share my thoughts regarding this project. :+1:

**Comment by KVGarg** [comment]
My thoughts on how can we can achieve this and how can we enhance coala-html:

1. Since `coala --json` outputs a result in `JSON` format in the terminal, using this functionality we can develop a static website in python using these results.
2. To enhance coala-html, we can
   
    -  provide a drop-down with Bears that are mentioned in `.coafile` ( + count warnings beside the bear-name) in the dropdown list
    - show an analytical (line or bar) graph that shows the number of warnings for each bear
    -  add an option for user to choose one or multiple bears from a list (an another drop-down other than the above one that will show only those bears that are not mentioned in `.coafile`) and ask user whether interested in re-running the analysis of code on the basis of selected bears. If yes, the website will be regenerated with new (or can append these new results to old results) results **(No idea whether this is achievable or not)**

thinking more on how can we enhance the `coala-html`, will keep adding side-by-side in future :innocent:

**Comment by AkshJain99** [comment]
@li-boxuan are you going to mentor this one in gsoc 19?

**Comment by li-boxuan** [comment]
@AkshJain99 Nope. I probably mentor this project: https://github.com/coala/projects/pull/704
